java/src/main/native/src/TableJni.cpp

```diff
@@ -4643,42 +4643,78 @@ Java_ai_rapids_cudf_Table_contiguousSplitGroups(JNIEnv* env,
 
     // 1) Gets the groups(keys, offsets, values) from groupby.
     //
-    // Uses only the non-key columns as the input values instead of the whole table,
-    // to avoid duplicated key columns in output of `get_groups`.
-    // The code looks like a little more complicated, but it can reduce the peak memory.
-    auto num_value_cols = input_table->num_columns() - key_indices.size();
-    std::vector<cudf::size_type> value_indices;
-    value_indices.reserve(num_value_cols);
-    // column indices start with 0.
-    cudf::size_type index = 0;
-    while (value_indices.size() < num_value_cols) {
-      if (std::find(key_indices.begin(), key_indices.end(), index) == key_indices.end()) {
-        // not key column, so adds it as value column.
-        value_indices.emplace_back(index);
+    // If the `jprojection_column_indices` is null, uses all_columns as projection columns;
+    // If the `jprojection_column_indices` is not null, use the `jprojection_column_indices` columns
+    // as projection columns.
+    std::vector<cudf::size_type> projection_indices;
+    auto num_project_cols = [&]() -> size_t {
+      if (jprojection_column_indices == NULL) {
+        // if a column is not in key columns, then it's a projection column
+        auto num_non_key_cols =
+          static_cast<size_t>(input_table->num_columns()) - key_indices.size();
+        if (num_non_key_cols < 0) {
```

Comment:
The line above it will only underflow as the type is `size_t` which is unsigned. We should check the two numbers before doing subtraction.